### ISSUE TITLE
fix: hidden menu items aren't skipped when using keyboard focus

### ISSUE THREAD

**Issue by caseyfaber** [microsoft/fast#5691]
# 🐛 Bug Report

If you use the `hidden` attribute on a menu item, and then use the up and down arrow keys to navigate through the menu items, the tab index of the hidden item will update. The focused item just stays on the previous one until down arrow is pushed again to get to the next item.

## 💻 Repro or Code Sample

https://codesandbox.io/s/menu-focuses-hidden-menu-items-oxdd0t

```html
<fast-menu>
  <fast-menu-item>Menu item 1</fast-menu-item>
  <fast-menu-item hidden>Menu item 2</fast-menu-item>
  <fast-menu-item>Menu item 3</fast-menu-item>
</fast-menu>
```
1. Press tab to focus on the first menu item.
2. Press down arrow.

## 🤔 Expected Behavior

I think focus should change to "Menu item 3".

## 😯 Current Behavior

"Menu item 1" stays focused. (And "Menu item 2" is the current item with `tabindex="0"`).

## 🔦 Context

I am making a menu that sometimes has the menu items filtered down, by using the `hidden` attribute on some of them.

## 🌍 Your Environment

* OS & Device: Windows 10
* Browser: Chrome
* Version: fast-foundation 2.33.4

**Comment by chrisdholt** [comment]
Thanks for the issue @caseyfaber, I'll leave the issue open for you to validate, but I believe this is expected and so I think the correct resolution would be to close this as by-design. While often times it's expected to skip disabled controls in the tabindex, there are a few scenarios where keeping them focusable is not only acceptable but expected. Menu items within a menu or menubar are one such scenario. Including a link to the relevant portion of the ARIA practices documentation here (number 2 in the list): https://www.w3.org/TR/wai-aria-practices-1.2/#kbd_disabled_controls.

**Comment by chrisdholt** [comment]
Definitely a bug and definitely an a11y issue. Thanks for the repro - it was extremely helpful to illustrate the situation!

**Comment by chrisdholt** [comment]
I have an assumption that this is likely the case with multiple elements. My thought is that we can make a quick fix here and look at a larger approach/pattern for resolving this. I can't think of a scenario where having hidden elements in the tabindex is acceptable so it should be a relatively quick and easy thing to resolve.

**Comment by chrisdholt** [comment]
This should publish in the release tonight, thanks @caseyfaber for this issue!
